dvc/objects/hash.py

```diff
@@ -16,52 +20,101 @@
     from .state import StateBase
 
 
-def _adapt_info(info: Dict[str, Any], scheme: str) -> Dict[str, Any]:
-    if scheme == "s3" and "ETag" in info:
-        info["etag"] = info["ETag"].strip('"')
-    elif scheme == "gs" and "etag" in info:
-        import base64
-
-        info["etag"] = base64.b64decode(info["etag"]).hex()
-    elif scheme.startswith("http") and (
-        "ETag" in info or "Content-MD5" in info
-    ):
-        info["checksum"] = info.get("ETag") or info.get("Content-MD5")
-    return info
-
-
 def dos2unix(data: bytes) -> bytes:
     return data.replace(b"\r\n", b"\n")
 
 
-def _fobj_md5(
-    fobj: BinaryIO,
-    hash_md5: "hashlib._Hash",
-    binary: bool,
-    chunk_size: int = 2**20,
-) -> None:
+def get_hasher(name: str) -> "hashlib._Hash":
+    try:
+        return getattr(hashlib, name)()
+    except AttributeError:
+        return hashlib.new(name)
+
+
+class HashedStreamReader(io.IOBase):
```

Comment:
I'd name this `HashFile` but that's already taken. If you have any better names, please do suggest.

Reply:
Okay, renamed into `HashStreamFile`.